traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -379,6 +379,23 @@ func (cg *TOCacheGroup) deleteCoordinate(coordinateID int) error {
 	return nil
 }
 
+func GetCacheGroupByName(name string, apiInfo *api.APIInfoImpl) (*tc.CacheGroupNullable, error) {
```

Comment:
I think we probably want to validate the existence of all the given cachegroups in a topology with a single DB query rather than one per cachegroup, since that can easily translate into tens of queries just for validation. So, this function as-is probably isn't necessary.

As a side note, I'm not a big fan of how this function uses the `APIInfoImpl` and temporarily swaps in params. It seems like a misuse of `APIInfoImpl` and seems fragile, but I can see why you're doing it. We really need to refactor the DB layer (or lack thereof) in Traffic Ops and reduce the amount of DB entrypoints all over the place.

Reply:
Getting all cachegroups in a single query in 4146f5b886 and no longer using `APIInfoImpl` that way.